From d96248554427a6806caf24d52bd86950af92fd77 Mon Sep 17 00:00:00 2001
From: LuK1337 <priv.luk@gmail.com>
Date: Thu, 19 Apr 2018 01:14:21 +0200
Subject: [PATCH 08/10] SystemUI: Fix navigation bar arrows visibility handling

* Fixes issue where input events would be sent
  while dpad_group view was set to be invisible.

Change-Id: Ie802e9c8157505a2ad706332f0b2fe715caf428c
---
 .../res/layout-sw600dp/nav_buttons_dpad_group_rot90.xml  | 5 +++--
 packages/SystemUI/res/layout/nav_buttons_dpad_group.xml  | 5 +++--
 .../SystemUI/res/layout/nav_buttons_dpad_group_rot90.xml | 5 +++--
 .../systemui/statusbar/phone/NavigationBarView.java      | 9 ++++-----
 4 files changed, 13 insertions(+), 11 deletions(-)

diff --git a/packages/SystemUI/res/layout-sw600dp/nav_buttons_dpad_group_rot90.xml b/packages/SystemUI/res/layout-sw600dp/nav_buttons_dpad_group_rot90.xml
index 38c0df4c0d7..5f9383c5686 100644
--- a/packages/SystemUI/res/layout-sw600dp/nav_buttons_dpad_group_rot90.xml
+++ b/packages/SystemUI/res/layout-sw600dp/nav_buttons_dpad_group_rot90.xml
@@ -19,8 +19,7 @@
     android:id="@+id/dpad_group"
     android:layout_width="fill_parent"
     android:layout_height="fill_parent"
-    android:clipChildren="false"
-    android:visibility="gone">
+    android:clipChildren="false">
 
         <com.android.systemui.statusbar.policy.KeyButtonView
             android:id="@+id/dpad_left"
@@ -30,6 +29,7 @@
             android:src="@drawable/ic_sysbar_ime_left"
             systemui:keyCode="21"
             systemui:keyRepeat="true"
+            android:visibility="gone"
             android:scaleType="center"
             android:contentDescription="@string/accessibility_dpad_left" />
 
@@ -41,6 +41,7 @@
             android:src="@drawable/ic_sysbar_ime_right"
             systemui:keyCode="22"
             systemui:keyRepeat="true"
+            android:visibility="gone"
             android:scaleType="center"
             android:contentDescription="@string/accessibility_dpad_right" />
 
diff --git a/packages/SystemUI/res/layout/nav_buttons_dpad_group.xml b/packages/SystemUI/res/layout/nav_buttons_dpad_group.xml
index 2f41a940d8c..d7bb9429fad 100644
--- a/packages/SystemUI/res/layout/nav_buttons_dpad_group.xml
+++ b/packages/SystemUI/res/layout/nav_buttons_dpad_group.xml
@@ -19,8 +19,7 @@
     android:id="@+id/dpad_group"
     android:layout_width="fill_parent"
     android:layout_height="fill_parent"
-    android:clipChildren="false"
-    android:visibility="gone">
+    android:clipChildren="false">
 
         <com.android.systemui.statusbar.policy.KeyButtonView
             android:id="@+id/dpad_left"
@@ -30,6 +29,7 @@
             android:src="@drawable/ic_sysbar_ime_left"
             systemui:keyCode="21"
             systemui:keyRepeat="true"
+            android:visibility="gone"
             android:scaleType="center"
             android:contentDescription="@string/accessibility_dpad_left" />
 
@@ -41,6 +41,7 @@
             android:src="@drawable/ic_sysbar_ime_right"
             systemui:keyCode="22"
             systemui:keyRepeat="true"
+            android:visibility="gone"
             android:scaleType="center"
             android:contentDescription="@string/accessibility_dpad_right" />
 
diff --git a/packages/SystemUI/res/layout/nav_buttons_dpad_group_rot90.xml b/packages/SystemUI/res/layout/nav_buttons_dpad_group_rot90.xml
index 38c0df4c0d7..5f9383c5686 100644
--- a/packages/SystemUI/res/layout/nav_buttons_dpad_group_rot90.xml
+++ b/packages/SystemUI/res/layout/nav_buttons_dpad_group_rot90.xml
@@ -19,8 +19,7 @@
     android:id="@+id/dpad_group"
     android:layout_width="fill_parent"
     android:layout_height="fill_parent"
-    android:clipChildren="false"
-    android:visibility="gone">
+    android:clipChildren="false">
 
         <com.android.systemui.statusbar.policy.KeyButtonView
             android:id="@+id/dpad_left"
@@ -30,6 +29,7 @@
             android:src="@drawable/ic_sysbar_ime_left"
             systemui:keyCode="21"
             systemui:keyRepeat="true"
+            android:visibility="gone"
             android:scaleType="center"
             android:contentDescription="@string/accessibility_dpad_left" />
 
@@ -41,6 +41,7 @@
             android:src="@drawable/ic_sysbar_ime_right"
             systemui:keyCode="22"
             systemui:keyRepeat="true"
+            android:visibility="gone"
             android:scaleType="center"
             android:contentDescription="@string/accessibility_dpad_right" />
 
diff --git a/packages/SystemUI/src/com/android/systemui/statusbar/phone/NavigationBarView.java b/packages/SystemUI/src/com/android/systemui/statusbar/phone/NavigationBarView.java
index 1be2205463a..42b743d2d98 100644
--- a/packages/SystemUI/src/com/android/systemui/statusbar/phone/NavigationBarView.java
+++ b/packages/SystemUI/src/com/android/systemui/statusbar/phone/NavigationBarView.java
@@ -854,12 +854,11 @@ public class NavigationBarView extends FrameLayout implements PluginListener<Nav
     }
 
     public void updateDpadKeys() {
-        if (mShowDpadArrowKeys) { // overrides IME button
-            final boolean showingIme = ((mNavigationIconHints
-                    & StatusBarManager.NAVIGATION_HINT_BACK_ALT) != 0);
+        final int visibility = mShowDpadArrowKeys && (mNavigationIconHints
+                & StatusBarManager.NAVIGATION_HINT_BACK_ALT) != 0 ? View.VISIBLE : View.GONE;
 
-            getDpadView().setVisibility(showingIme ? View.VISIBLE : View.INVISIBLE);
-        }
+        getDpadView().findViewById(R.id.dpad_left).setVisibility(visibility);
+        getDpadView().findViewById(R.id.dpad_right).setVisibility(visibility);
     }
 
     private final Consumer<Boolean> mDockedListener = exists -> mHandler.post(() -> {
-- 
2.17.0

